### ISSUE TITLE
XQueries for works

### ISSUE THREAD

**Issue by wsalesky** [srophe/srophe-eXist-app#414]
Abbreviations and Editions - reconcile. 
Add related persons to works and related works to persons.

**Comment by davidamichelson** [comment]
This example is based on http://wwwb.library.vanderbilt.edu/exist/apps/srophe/work/250
Please use REGEX and XQuery to make the following changes:

For each record that has content in the text node of /TEI/text/body/bibl/note/@type[editions]

Please make the following changes:

Original of /TEI/text/body/bibl/note/@type[MSS]
Reads: ```<note xml:lang="en" type="MSS" source="#bib250-1">
               <bibl>ed (1): coll. sur copie d’un ms de Londres par Wright + variantes du ms Berlin, Königliche Bibliothek, syr. 75 (Sachau 222), f. 260-271</bibl>
            </note>```
It should be changed to:

**Comment by davidamichelson** [comment]
Once this is in place we will have to visualize it but we can open a new ticket for that ;)

**Comment by wsalesky** [comment]
What about a note with this content:
ed (1): = ed (2) + collation du ms Vatican, Syr. 117, f. 548v-550v par Guidi

See: http://wwwb.library.vanderbilt.edu/exist/apps/srophe/work/352

**Comment by davidamichelson** [comment]
Ha ha! Great, I was not aware of that. I'll work on it.

**Comment by wsalesky** [comment]
Just popped up in my first test case. :)

**Comment by davidamichelson** [comment]
Okay, so just treat those the same way as everything else. There are only a few and we'll fix them by hand. So for now just use REGEX to remove the first part "ed (1): ".

The new text node should read "= ed (2) + collation du ms Vatican, Syr. 117, f. 548v-550v par Guidi" we will use the "=" to find and hand fix these.

**Comment by wsalesky** [comment]
Would the ideal solution be to replace both with the above xml? I think I can do this using regex.

**Comment by davidamichelson** [comment]
Note, in a very few cases the element /TEI/text/body/bibl/note/@type[modernTranslation] also has some edition number information but don't touch those or include them in the transformation. There are only a dozen and we will do them by hand as they don't match in a regular way.

**Comment by davidamichelson** [comment]
I don't follow the question, can you rephrase please?

**Comment by wsalesky** [comment]
Sorry, in this case:
ed (1): = ed (2) + collation du ms Vatican, Syr. 117, f. 548v-550v par Guidi

Should the bibl output be:
```
<bibl xml:id="witness1"> = <bibl xml:id="witness2"> + collation du ms Vatican, Syr. 117, f. 548v-550v par Guidi
```

**Comment by davidamichelson** [comment]
No, I would say for now just leave the "ed (2) + ..." in the text node.

But if it is possible it would be nice to wrap "ed (2)" with a ```<ref
target="#witness2"> that would be a great help. Otherwise we can do that by
hand.

**Comment by wsalesky** [comment]
Okay, I ran the script for this one, we will need to do some additional spot checking.

**Comment by davidamichelson** [comment]
Great!

**Comment by wsalesky** [comment]
@davidamichelson Thinking about where to store these queries, it is useful to keep them even if they are essentially only one-off scripts. Thinking I can store them in the data app or the admin module. The admin module is not currently on github. Thoughts?

**Comment by davidamichelson** [comment]
Hmm, okay I did the spot checking on this and found a couple of problems. One is in the xquery and one may just be in the visualization. These errors can be seen here: http://wwwb.library.vanderbilt.edu/exist/apps/srophe/work/286

1. It looks like maybe my comment above was unclear. We want to remove any leading edition numbers from the text node: "Ed(1):" etc. from each record. But it appears that this text has not been stripped out. My comment above "for now just leave the "ed (2) + ..." in the text node." was only for when there are more than one edition number present.

2. The ordering and numbering of manuscripts is not rendering correctly in the HTML. In the list of manuscripts in the HTML http://wwwb.library.vanderbilt.edu/exist/apps/srophe/work/286 the manuscripts with ```<note xml:lang="en" type="MSS"><bibl xml:id="witness1">``` should appear as number one in the list and ```<note xml:lang="en" type="MSS"><bibl xml:id="witness2">``` should appear as number two, etc. Then we should number all the other note elements without a bibl@xml:id as ```<note xml:lang="en" type="MSS"><bibl xml:id="witness3">``` etc.

Please let me know if that makes sense and thanks!

**Comment by wsalesky** [comment]
Ah, I think I matched on the pattern ed:(2) and ed: (2). I will have to rerun the script.

**Comment by wsalesky** [comment]
@davidamichelson So all the notes with type='MSS' should have an xml:id?

**Comment by davidamichelson** [comment]
Yes please, that way they each have a witnessid if we want to use them.

**Comment by wsalesky** [comment]
Done. Need to be reviewed before I close the issue.

**Comment by davidamichelson** [comment]
Please add a test for records where editions or translations do not have a ref to a MSS such as http://wwwb.library.vanderbilt.edu/work/1200 so as to supress the text: (Edition from manuscript witness . See below.)

**Comment by davidamichelson** [comment]
Please check the XSLT for Editions and MSS Atr. Corresp 
Example: http://wwwb.library.vanderbilt.edu/exist/apps/srophe/work/250 is not rending the relationships

**Comment by davidamichelson** [comment]
@wsalesky Alas, the fix did not work because the numbers of the new bibl-ids and the auto numbering on the page do not match: see http://wwwb.library.vanderbilt.edu/exist/apps/srophe/work/250 where MS Witness 1 is the same as bib250-19 but numbered as 1 in the MSS list and 19 in editions.

**Comment by wsalesky** [comment]
Darn! I will take a look and rerun.

**Comment by wsalesky** [comment]
Well, there is a problem with the xslt, not the xml. Sort of. The ids in the xml matches up, but the display does not, because the bibl # is based on all the other bibls, which of course messes up the MSS count. I didn't think of that when I transformed the script.
